src/t_string.c

```diff
@@ -672,10 +674,13 @@ void incrbyfloatCommand(client *c) {
         return;
     }
     new = createStringObjectFromLongDouble(value,1);
+    dictEntry *de;
     if (o)
-        dbReplaceValue(c->db,c->argv[1],new);
+        de = dbReplaceValue(c->db, c->argv[1], new);
     else
-        dbAdd(c->db,c->argv[1],new);
+        de = dbAdd(c->db,c->argv[1],new);
```

Comment:
I think this is a good example of the new robj ownership model. This code is written to have minimal delta from the previous code, but also makes the code very weird. Since dbReplaceValue (among other functions) now consumes the input, it means a lot of the surrounding code no longer makes sense.

We would probably want to remove all of the dbReplaceValue calls and instead directly call functions that create and embed the object. eg:
```
new = createAndAddStringObjectFromLongLongDouble(db, key, value, double);
```
This just seems like a lot more refactoring. 

Another thought is it might also make sense to implement some type of "unlinked redisObject", which is a dictEntry without a key or next pointer, that could be hooked up on add. That resolves some of these cases where we are creating an object before inserting it.